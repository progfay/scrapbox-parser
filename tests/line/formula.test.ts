/* global describe it expect */

import '../jest-setup'

describe('formula', () => {
  it('Simple formula', () => {
    expect('[$ \\frac{3}{2}^N]').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'formula',
            formula: '\\frac{3}{2}^N'
          }
        ]
      }
    ])
  })

  it('Formula includes [] with tail half-space', () => {
    expect('[$ [x] ]').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'formula',
            formula: '[x]'
          }
        ]
      }
    ])
  })

  it('Formula includes [] without tail half-space', () => {
    expect('[$ [x]]').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'formula',
            formula: '[x'
          },
          {
            type: 'plain',
            text: ']'
          }
        ]
      }
    ])
  })
})
