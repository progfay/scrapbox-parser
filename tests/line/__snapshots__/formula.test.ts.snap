// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`formula > Formula includes [] with tail half-space 1`] = `
[
  {
    "indent": 0,
    "nodes": [
      {
        "formula": "[x]",
        "raw": "[$ [x] ]",
        "type": "formula",
      },
    ],
    "type": "line",
  },
]
`;

exports[`formula > Formula includes [] without tail half-space 1`] = `
[
  {
    "indent": 0,
    "nodes": [
      {
        "formula": "[x",
        "raw": "[$ [x]",
        "type": "formula",
      },
      {
        "raw": "]",
        "text": "]",
        "type": "plain",
      },
    ],
    "type": "line",
  },
]
`;

exports[`formula > Simple formula 1`] = `
[
  {
    "indent": 0,
    "nodes": [
      {
        "formula": "\\frac{3}{2}^N",
        "raw": "[$ \\frac{3}{2}^N]",
        "type": "formula",
      },
    ],
    "type": "line",
  },
]
`;
